app/users/edit/credits/controller.js

```diff
@@ -0,0 +1,69 @@
+import Controller from '@ember/controller'
+import { task } from 'ember-concurrency'
+import QueryParams from 'ember-parachute'
+import moment from 'moment'
+
+const UsersEditCreditsQueryParams = new QueryParams({
+  year: {
+    defaultValue: moment().year(),
+    replace: true,
+    refresh: true
+  }
+})
+
+export default Controller.extend(UsersEditCreditsQueryParams.Mixin, {
+  years: task(function*() {
+    let employments = yield this.store.query('employment', {
+      user: this.get('model.id'),
+      ordering: 'start_date'
+    })
+
+    let from = (employments.get('firstObject.start') || moment()).year()
+    let to = moment()
+      .add(1, 'year')
+      .year()
+
+    return [
+      ...[...new Array(to + 1 - from).keys()].map(i => {
+        let y = from + i
+
+        return { value: y, label: y }
+      }),
+      { value: '', label: 'All' }
+    ]
+  }),
+
+  setup() {
+    this.get('years').perform()
+    this.get('absenceCredits').perform()
+    this.get('overtimeCredits').perform()
+  },
+
+  queryParamsDidChange({ shouldRefresh }) {
+    if (shouldRefresh) {
+      this.get('absenceCredits').perform()
+      this.get('overtimeCredits').perform()
+    }
+  },
+
+  absenceCredits: task(function*() {
+    let year = this.get('year')
+
+    return yield this.store.query('absence-credit', {
+      user: this.get('model.id'),
+      include: 'absence_type',
+      ordering: '-date',
+      ...(year ? { year } : [])
```

Comment:
This looks crazy, but i guess it makes sense :smile: One question though: Can year be undefined when you define a defaultValue?

Reply:
It can be whatever you set it. In this case, I set it to explicitly to `null` which is why I do this thingy there..

Reply:
Actually this should be `(year ? { year } : {})`, which would make it more readable!